### PR TITLE
[Merged by Bors] - feat: a type with a non trivial filter has to be nonempty

### PR DESCRIPTION
**Pull request by ADedecker**
Preliminaries for #6844 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Filter/Basic.lean [hunk 1]
```diff
@@ -776,6 +776,9 @@ instance unique [IsEmpty α] : Unique (Filter α) where
   uniq := filter_eq_bot_of_isEmpty
 #align filter.unique Filter.unique
 
+theorem NeBot.nonempty {f : Filter α} (hf : NeBot f) : Nonempty α :=
```
Comment (ocfnash) [inline Mathlib/Order/Filter/Basic.lean:779]: Have you considered:
```suggestion
theorem NeBot.nonempty (f : Filter α) [f.NeBot] : Nonempty α :=
```
It's not clear to me which is preferable but if you have an application to hand, then perhaps you can tell.

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Order/Filter/Basic.lean (hunk 1 above)
Comment (ADedecker) [reply Mathlib/Order/Filter/Basic.lean:779]: I actually have no application :facepalm: I probably used it in a previous iteration of Ascoli but don't remember how... Let's do your version then.

**Comment by ADedecker** [comment]
bors merge

**Pull request closed without merging**